src/main/java/uk/gov/justice/framework/tools/command/SearchLogs.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.justice.framework.tools.command;
+
+import uk.gov.justice.framework.tools.common.command.ShellCommand;
+import uk.gov.justice.log.search.KibanaQueryBuilder;
+import uk.gov.justice.log.search.SearchCriteria;
+import uk.gov.justice.log.search.SearchService;
+import uk.gov.justice.log.search.main.output.FilePrinter;
+import uk.gov.justice.log.search.main.output.HTMLPrinter;
+import uk.gov.justice.log.search.main.output.OutputPrinter;
+import uk.gov.justice.log.utils.ConnectionManager;
+import uk.gov.justice.log.utils.PropertyReader;
+import uk.gov.justice.log.utils.RestConfig;
+import uk.gov.justice.log.utils.SearchParameters;
+import uk.gov.justice.log.wrapper.HttpAsyncClientBuilderWrapper;
+import uk.gov.justice.log.wrapper.RequestConfigBuilderWrapper;
+import uk.gov.justice.log.wrapper.RestClientFactory;
+
+import java.io.IOException;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.jayway.jsonpath.JsonPath;
+import net.minidev.json.JSONArray;
+import org.apache.http.util.EntityUtils;
+import org.elasticsearch.client.Response;
+import org.elasticsearch.client.RestClient;
+
+public class SearchLogs extends AbstractLogSearcherCommand implements ShellCommand {
+
+    protected KibanaQueryBuilder kibanaQueryBuilder;
+    private SearchService searchService;
+    private SearchCriteria searchCriteria;
+    private RestConfig restConfig;
+    private SearchParameters searchParameters;
+
+    public SearchLogs(final KibanaQueryBuilder kibanaQueryBuilder) {
+        this.kibanaQueryBuilder = kibanaQueryBuilder;
+        this.searchService = new SearchService(restClient());
+    }
+
+    @VisibleForTesting
+    public SearchLogs(final KibanaQueryBuilder kibanaQueryBuilder, final SearchService searchService) {
+        this.kibanaQueryBuilder = kibanaQueryBuilder;
+        this.searchService = searchService;
+    }
+
+    @Override
+    public void run(String[] strings) {
+        try {
+            this.searchParameters = new SearchParameters(strings[0], strings[1], strings[2], strings[3]);
+            final PropertyReader propertyReader = new PropertyReader(searchParameters);
+            if (propertyReader.errors().size()>0){
+                throw new IllegalArgumentException(""+ propertyReader.errors());
+            }
+            this.restConfig = propertyReader.restConfig();
+            this.searchCriteria = propertyReader.searchCriteria();
+            final Response response = searchService.search(kibanaQueryBuilder);
+            printResults(kibanaQueryBuilder, response);
+        } catch (IOException e) {
+            consolePrinter.writeException(e);
+        }
+    }
+
+    private void printResults(final KibanaQueryBuilder kibanaQueryBuilder, final Response response) {
```

Comment:
This class is already doing a lot - perhaps output needs to be delegated completely to another class?